### PR TITLE
Fix bug in HTTP Server with Sequences

### PR DESCRIPTION
**Pull request by HonakerM**
**What this PR does / why we need it**:
This PR fixes issue #532. There was a slight regression with the HTTP server that this fixes and adds a test case to ensure this doesn't happen in the future.

**If applicable**:
- [ ] this PR contains documentation
- [ ] this PR contains unit tests
- [ ] this PR has been tested for backwards compatibility

### CODE REVIEW

File: tests/fixtures/sample_lib/data_model/sample.py [hunk 1]
```diff
@@ -103,6 +103,19 @@ class FileTask(TaskBase):
...
+    required_parameters={
+        "str_type": str,
+        "list_str_type": List[str],
+        "int_type": int,
+        "list_int_type": List[int],
+    },
+    unary_output_type=StrSequence,
+)
+class PrimitiveTask(TaskBase):
+    """A sample task for testing ra"""
```
Comment (joerunde) [inline tests/fixtures/sample_lib/data_model/sample.py:116]: What's `ra`?

Comment (HonakerM) [reply tests/fixtures/sample_lib/data_model/sample.py]: 🤦 a spelling mistake. Just fixed it

File: tests/fixtures/sample_lib/modules/primitive/primitive_module.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+from ...data_model.sample import PrimitiveTask
+from caikit.core.modules import ModuleLoader
+from caikit.interfaces.common.data_model import StrSequence
+import caikit.core
+
+
+@caikit.core.module(
+    "bd87fbbe-caad-4db6-93f5-37cc3dbedef5", "PrimitiveModule", "0.0.1", PrimitiveTask
+)
+class PrimitiveModule(caikit.core.ModuleBase):
```
Comment (joerunde) [inline tests/fixtures/sample_lib/modules/primitive/primitive_module.py:31]: @HonakerM Does this problem only occur if the `List[T]` types are in the task's set of required parameters? We do have a `sample_lib.modules.sample_task.SamplePrimitiveModule` already which has a bunch of primitive types including `List[T]` and `Union[List[T]...]`, would that one work as is if we called it via the http server?

Comment (HonakerM) [reply tests/fixtures/sample_lib/modules/primitive/primitive_module.py:31]: @joerunde I missed that one! It should be testing the same stuff so I'll remove my custom module.

**Review by joerunde — APPROVED** [review]
get it!

**Pull request merged**